//------------------------------------------------------------------------------
// <auto-generated>
//     This code was generated by AsyncGenerator.
//
//     Changes to this file may cause incorrect behavior and will be lost if
//     the code is regenerated.
// </auto-generated>
//------------------------------------------------------------------------------


using System;
using System.Linq;
using System.Threading;
using System.Threading.Tasks;
using AsyncGenerator.TestCases;

namespace AsyncGenerator.Tests.ExceptionHandling.Input
{
	public partial class PropagateOperationCanceledException
	{
		public async Task MethodThatCatchesExceptionsAsync(CancellationToken cancellationToken)
		{
			try
			{
				await (SimpleFile.ReadAsync(cancellationToken));
			}
			catch (OperationCanceledException) { throw; }
			catch (Exception ex)
			{
			    throw new Exception("My wrapped exception", ex);
			}
		}
	}
}